### ISSUE TITLE
💡 Synopsis in general /episodes endpoint

### ISSUE THREAD

**Issue by Fribb** [jikan-me/jikan-rest#341]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

Currently, to read the synopsis of an episode you have to request that specific episode. This leads to a lot of requests because each episode has to be requested individually.

The only difference between an episode element in /episodes and /episode/1 is the available synopsis element.

### Describe the solution you'd like

If available, the synopsis should be available as well in the /episodes endpoint to cut down the potential requests to be made to the API. 

It would also be fantastic if the /episodes endpoint could trigger the update of the episodes metadata

### Teachability, Documentation, Adoption, Migration Strategy

As described above, it would cut down the number of requests a user has to do if they don't have the ability to update individual episode metadata. You would be able to hit up the /episodes endpoint and go from there which would reduce the number of requests made to the API by once for every 100 episodes instead of X times for all episodes.

**Comment by Fribb** [comment]
I updated the initial enhancement with further information.

Apparently, the `/episodes` and `/episodes/X` endpoints provide different sets of information as you can see in the provided table.

This would mean that to get a comprehensive result you would not only request the individual episodes but also the episode list with each page. 

So an Anime like One Piece would require:

* Currently 1065 individual Episode requests
* Currently 11 `/episodes?page=X` requests

So 1076 requests in total, according to the [v4 rate limit](https://docs.api.jikan.moe/#section/Information/Rate-Limiting) with 3 Requests each second this would take roughly 358 Seconds (roughly 6 Minutes) to request.

Consolidating the Information on the `/episodes` endpoint would, as initially stated, cut down this time to 3 seconds.

**Comment by pushrbx** [comment]
The API is following what is visible to the user when yourself go on an anime's datasheet:
- You can see "episodes" tab on the page
- If you click on it you will get a list of it
- If you click on an item in the list you get more info about it.

If Jikan would want to consolidate it into one, it would require to regularly scrape all anime data-sheets to have an up to date database about the episodes, which is not feasible to do. (Also it's not fair towards MAL)

If your request to /episode and /episodes is not yet cached, it translates to an http request to MAL and the scraper parses the page, then you get the scraper's result.

**Comment by Fribb** [comment]
I can understand that the update process could be a bit much, especially with data that hasn't been cached yet. With slow connections or anything like that queuing 100 requests to update the individual episodes might end more in a timeout.

But providing the information on an `/episodes` level would still be beneficial even if it is just outdated data maybe as a parameter.

Still, I don't really think that it is unfair towards MAL. An API should make things easier and faster to use for someone who wants to consume the data.

Correct me if I'm wrong but wasn't v4 intended to have a better data structure to not be as static as v3 was? I remember asking the same thing a while ago but at that time this was basically the answer that this wasn't possible with v3 and that it might be with v4 by joining the tables.

**Comment by irfan-dahir** [comment]
> Correct me if I'm wrong but wasn't v4 intended to have a better data structure to not be as static as v3 was? I remember asking the same thing a while ago but at that time this was basically the answer that this wasn't possible with v3 and that it might be with v4 by joining the tables.

Yes, v4 took that step towards resources (Anime, Manga, etc) and search. 

We can look into the feasibility of this in the next phase of things. At the moment it's not pushing the episodes+details into a table. It's just caching the JSON response into the DB - like a lot of the other requests which are parsed directly. 

I understand the problem and do agree with you. But this would require some major overhaul to how things are processed, stored, and served so wouldn't say this would be something done in the near future.
